pkg/runner/context.go

```diff
@@ -203,6 +244,20 @@ func (c *ExecutionContext) GenerateEnvfile() error {
 		logrus.Fatalf("Error writing out file: %s\n", err.Error())
 	}
 
+	// wait for the file to exist before continuing
+	// this is to address an issue where the file does not exist on disk before the command
+	// is executed
+	for {
+
+		if utils.FileExists(c.Envfile.Path) {
+			logrus.Debug(fmt.Sprintf("File has been found: %s\n", c.Envfile.Path))
+			break
+		}
+
+		// sleep for 1ms
+		time.Sleep(1 * time.Millisecond)
+	}
+
```

Comment:
This is never reached if the file fails to write on L:244  - log.Fatalf() => writes a log and exits with 1, so no need for this. 

This most probably sounds like a race condition somewhere... 

also if possible I would try to avoid `time.Sleep()` statements

Reply:
OK. I spoke to Jack last night and he explained the what the overall issue is. Am going to remove this and change the way that the file is created, so it can be named properly and not clash with other parallel tasks.